opentelemetry-sdk/src/opentelemetry/sdk/_logs/__init__.py

```diff
@@ -319,9 +321,33 @@ def __init__(
 
     @staticmethod
     def _get_attributes(record: logging.LogRecord) -> Attributes:
-        return {
+        attributes = {
             k: v for k, v in vars(record).items() if k not in _RESERVED_ATTRS
         }
+        if record.exc_info:
+            exctype, value, tb = record.exc_info
+            if exctype is not None:
+                exc_type = exctype.__name__
+            if value is not None:
+                message = value.args[0]
+            callstack = []
+            if tb is not None:
+                for file_name, line, method, text in traceback.extract_tb(tb):
+                    callstack.append(
+                        {
+                            "method": method,
+                            "fileName": file_name,
+                            "line": line,
+                            "text": text,
+                        }
+                    )
+                callstack.reverse()
+            attributes[SpanAttributes.EXCEPTION_TYPE] = exc_type
+            attributes[SpanAttributes.EXCEPTION_MESSAGE] = message
+            attributes[SpanAttributes.EXCEPTION_STACKTRACE] = json.dumps(
+                callstack
+            )
```

Comment:
Use [`traceback.format_exception`](https://docs.python.org/3/library/traceback.html#traceback.format_exception)?

Reply:
I used this originally and it prints out the string in a very unformatted way.

```
        "exception.stacktrace": [
            "Traceback (most recent call last):\n",
            "  File \"<path>", line 38, in <module>\n    div = 1 / 0\n",            "ZeroDivisionError: division by zero\n"
        ]
```
Although we do have freedom for specifying whatever format we wish, doing this makes it difficult for the user to extract individual stacktrace information. Also, it is difficult to make sense of what it means.

Reply:
I believe we are doing similar string formatting in the tracing `record_exception` hence my inclination for consistency. but I like the line by line detailed stack trace list of {fileName, method, line, text}. If you look at what other SDKs do, it is mostly some form of string representation of exception. I would imagine some backends also use string datatype. now, I would like to know what do you think?

Reply:
The consistency is a good point. Yes most other SDKs do something similar to `traceback.format_exception`. I don't have a huge opinion here and don't really want to deal with inconsistency either so I'm okay with changing it to not use `json`.

Reply:
I have seen some backend systems that expect formatted stacktrace instead of json dump of {fileName, method, line, text}. This would be problematic for something that is built on top such as error aggregation system. I would suggest we go with the formatted representation. Also would like to know what do others think about this? @open-telemetry/python-approvers

Reply:
The specification has the expected representation: https://github.com/open-telemetry/opentelemetry-specification/blob/9fa7c656b26647b27e485a6af7e38dc716eba98a/specification/trace/semantic_conventions/exceptions.md#stacktrace-representation